src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs

```diff
@@ -376,6 +376,8 @@ public DSCPythonCodeCompletionProviderCore()
             BasicVariableTypes.Add(Tuple.Create(LIST_VARIABLE, typeof(PyList)));
             BasicVariableTypes.Add(Tuple.Create(DICT_VARIABLE, typeof(PyDict)));
 
+            Python.Included.Installer.SetupPython().Wait();
```

Comment:
why is this necessary? The DSCPython engine should do this already, should it not?
I think it's worth testing this change in Revit context if it's really required.

Reply:
There is a note in Dynamo
```
NOTE: Calling SetupPython multiple times will add the install location to the path many times,
        /// potentially causing the environment variable to overflow.
```
I think we might be doing this already ...maybe we should try to only run `SetupPython()` only once ...(keep a isAlreadySetup flag somewhere)
Please search for all occurrences of `SetupPython()`

Reply:
I actually was able to get in here without having python initialized. And we are already calling this method in multiple places. I will take a look.

Reply:
We are now calling SetupPython() every where. That method maintain a just do it once flag.